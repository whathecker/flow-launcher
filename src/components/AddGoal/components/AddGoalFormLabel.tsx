import React from "react";
import { Image, StyleSheet } from "react-native";
import { View, Text } from "../../Themed";
import { Container, Typography, Color } from "../../../styles";

type AddGoalFormLabelProps = {
  text: string;
  type: "motivation" | "goal";
};

const AddGoalFormLabel: React.FC<AddGoalFormLabelProps> = ({
  text,
  type,
}: AddGoalFormLabelProps) => {
  return (
    <View style={styles.labelWrapper}>
      <Text
        lightColor={Color.light.labelOnBackgroundForRead}
        darkColor={Color.dark.labelOnBackgroundForRead}
        style={styles.labelText}
      >
        {text}
      </Text>
      {type === "motivation" ? (
        <Image
          style={styles.icon}
          source={require(`../../../../assets/images/rocket_1f680.png`)}
        />
      ) : null}
      {type === "goal" ? (
        <Image
          style={styles.icon}
          source={require(`../../../../assets/images/sauropod_1f995.png`)}
        />
      ) : null}
    </View>
  );
};

const styles = StyleSheet.create({
  labelWrapper: {
    ...Container.flexStart,
    paddingBottom: 10,
  },
  labelText: {
    ...Typography.h1,
    fontSize: 20,
    paddingRight: 12,
  },
  icon: {
    width: 24,
    height: 24,
  },
});

export default AddGoalFormLabel;
