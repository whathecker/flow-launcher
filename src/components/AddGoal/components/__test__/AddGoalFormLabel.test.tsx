import * as React from "react";
import renderer from "react-test-renderer";
import AddGoalFormLabel from "../AddGoalFormLabel";

describe("Test AddGoalFormLabel component", () => {
  it("Should renders AddGoalFormLabel correctly", () => {
    const tree = renderer
      .create(<AddGoalFormLabel text="Motivation" type="motivation" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
