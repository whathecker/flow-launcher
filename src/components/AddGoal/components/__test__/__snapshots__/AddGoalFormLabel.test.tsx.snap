// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddGoalFormLabel component Should renders AddGoalFormLabel correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingBottom": 10,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#848484",
        },
        Object {
          "fontFamily": "Lexend-Bold",
          "fontSize": 20,
          "fontWeight": "bold",
          "paddingRight": 12,
        },
      ]
    }
  >
    Motivation
  </Text>
  <Image
    source={1}
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
</View>
`;
